### ISSUE TITLE
Rank conjunction expression returns incorrect result shape

### ISSUE THREAD

**Issue by bernecky** [Co-dfns/Co-dfns#433]
```
Dyalog APL/S-64 Version 18.0.39660
gcc (Ubuntu 10.2.0-13ubuntu1) 10.2.0
Co-dfns commit #: 0d40d11

      F←{⌽⍤1⊢2 3 4⍴⍳⍵}
      F 100
 3  2  1  0
 7  6  5  4
11 10  9  8
           
15 14 13 12
19 18 17 16
23 22 21 20
      ⍴F 100
2 3 4
      )cs
#
      ]compile ns1 modu -af=cpu
```
NB. LOTS of multi-line warnings, e.g., 

```
> modu.cpp:92:15: warning: deleting object of polymorphic class type ‘DOK’ which has non-virtual destructor might cause undef
>     92 | #define EX(x) delete x                                                                                             
>        |               ^~~~~~~~                                                                                             
>     93 | #define EF(init,ex,fun) EXPORT V ex##_dwa(lp*z,lp*l,lp*r){try{\                                                    
>        | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~    
> 
>      modu.F 100
>  3  2  1  0
>  7  6  5  4
> 11 10  9  8
> 15 14 13 12
> 19 18 17 16
> 23 22 21 20
>       ⍴modu.F 100
> 1 6 4         
```

NB. differences in result shapes.

**Comment by bernecky** [comment]
What is the approved method of automatically running all unit tests distributed with the compiler? I did not see
anything evident in the README file or in the docs folder.
Ditto for all unit tests (being able to select backend(s)), of course?

**Comment by arcfide** [comment]
See the util namespace. In short:

1. Make sure you're running with the development workspace found in the Co-dfns repository, not loading the namespace yourself or using the user command functionality.
2. `]cd` into the tests directory.
3. Use the `TEST` function. 

The TEST function can be called as `TEST'ALL'` to run all tests. 

Be advised, the Co-dfns project explicitly has no unit tests in it as a matter of principle and design/architecture. It does have a suite of integration/regression/black box tests. Additionally, the documentation policy of the Co-dfns project at the moment is not to include any developer level documentation in the README or the manuals, as those are user-level documents. Developers are expected to read the source code and understand the repository file hierarchy. This hierarchy and the source code interfaces *do* change on a regular basis (there is currently a major runtime architectural change happening as I write this), so keeping developer level documentation up to date is a waste of time right now, especially given that the interfaces are obvious by reading the code. 

Use the namespace interface mechanism of changing backends when you want to test different backends. 

-- 
Aaron W. Hsu | arcfide@sacrideo.us | http://www.sacrideo.us

**Comment by arcfide** [comment]
Thanks for the submission! 

-- 
Aaron W. Hsu | arcfide@sacrideo.us | http://www.sacrideo.us

**Comment by arcfide** [comment]
Fixed in v2.0.0. Thanks for the report!
